src/merkle/cmr.rs

```diff
@@ -51,7 +51,7 @@ impl CommitMerkleRoot for Cmr {
             CommitNodeInner::Witness => Cmr::tag_iv(b"Simplicity-Draft\x1fCommitment\x1fwitness"),
             CommitNodeInner::Fail(_, _) => Cmr::tag_iv(b"Simplicity-Draft\x1fCommitment\x1ffail"),
             CommitNodeInner::Hidden(h) => *h,
-            CommitNodeInner::Jet(jet) => jet.cmr(),
+            CommitNodeInner::Jet(j) => Cmr::tag_iv(b"Simplicity-Draft\x1fJet").update_1(j.cmr()),
```

Comment:
That line depends on the value of `jet.cmr()`. Originally, I intended this to include any IV, for efficiency reasons. Does the C code not include the IV?

Reply:
We don't have any static initializing, we are not gaining any efficiency. I did this because that's how the cmr_iv is used in imr calculation. Specifically, 
```rust
impl CommitMerkleRoot for Imr {
    fn get_iv<J: Jet>(node: &CommitNodeInner<J>) -> Self {
        match node {
            CommitNodeInner::Disconnect(_, _) => {
                Imr::tag_iv(b"Simplicity-Draft\x1fIdentity\x1fdisconnect")
            }
            CommitNodeInner::Witness => Imr::tag_iv(b"Simplicity-Draft\x1fIdentity\x1fwitness"),
            _ => Cmr::get_iv(node).into_inner().into(),
        }
    }
}
```

This would also require a jet branch which felt unclean to me. That's why included it here.

Reply:
I'm not sure if the Rust compiler is smart enough to convert [our current code](https://github.com/sanket1729/rust-simplicity/blob/update_tests_ffi/src/jet/init/core.rs#L445) into static return values. But, couldn't we change the code to a match statement that returns `Cmr(Midstate([...]))` in each branch? That is a value that is `const`, and the compiler should be smart enough to save this as a static in the program.